### PR TITLE
Removed Timer class, use clock() instead.

### PR DESCRIPTION
**Pull request by ctrl-z-9000-times**

### CODE REVIEW

**Comment by ctrl-z-9000-times** [comment]
We can replace the Timer class with calls to the clock() from <time.h>
```
clock_t start_time = clock();
elapsed_time = (double)(clock() - start_time) / CLOCKS_PER_SEC;
```
The clock() function returns the amount of CPU time that the calling process has consumed, which is *probably what we want for measuring speed performance*. Currently the timer measure the real system time.

We really don't need a whole class to make two standard library calls. But since we have a Timer class: we also have documentation and unit-tests for it. We have unit-tests for a standard library function ...

The method `float Timer::getSpeed()` can be moved to nupic/utils as a single function.

**Comment by ctrl-z-9000-times** [comment]
See issue #175

**Comment by ctrl-z-9000-times** [comment]
Timer is tested but not used by the python repo:
```
$ grep -rnl "Timer" nupic.py/
src/nupic/engine/__init__.py <= Exposes the Timer
tests/unit/nupic/engine/syntactic_sugar_test.py <= Tests Timer
```
Here is a place where I'd really like to clean up the python repo. Python, like C++, has a perfectly good standard library function for the time, `time.clock()` which is a wrapper around the `clock()` function that the C++ now uses.

File: src/test/unit/os/TimerTest.cpp [hunk 1]
```diff
@@ -1,96 +0,0 @@
...
-  ASSERT_GT(t1.getElapsed(), 0);
-  ASSERT_GT(t2.getElapsed(), t2elapsed);
-  ASSERT_GT(t2.getElapsed(), t1.getElapsed());
-
-  t1.start();
-  t1.stop();
-  ASSERT_EQ(t1.getStartCount(), 2u);
-}
-
-TEST(TimerTest, Drift) {
```
Comment (breznak) [inline src/test/unit/os/TimerTest.cpp]: you should keep an equivalent of this test somewhere, please.

File: src/test/unit/algorithms/SpatialPoolerTest.cpp [hunk 2]
```diff
@@ -1981,7 +1981,7 @@ TEST(SpatialPoolerTest, testSerialization2) {
     {
       SpatialPooler spTemp;
 
-      testTimer.start();
+      testTimer -= clock();
```
Comment (breznak) [inline src/test/unit/algorithms/SpatialPoolerTest.cpp]: use `getElapsed()` from the new utils/Time.hpp

File: src/test/unit/os/TimerTest.cpp (hunk 1 above)
Comment (breznak) [reply src/test/unit/os/TimerTest.cpp]: and a test that uses `sleep()` and timer to check it measures ok in getElapsed() .

File: src/nupic/engine/Region.hpp [hunk 3]
```diff
@@ -55,7 +55,7 @@ class Array;
 class Spec;
 class NodeSet;
 class BundleIO;
-class Timer;
+typedef clock_t Timer;
```
Comment (breznak) [inline src/nupic/engine/Region.hpp]: could this stay only in implementation .cpp?

File: src/nupic/engine/Region.cpp [hunk 4]
```diff
@@ -202,12 +202,12 @@ void Region::compute() {
...
 
   if (profilingEnabled_)
-    computeTimer_.start();
+    computeTimer_ -= clock();
 
   impl_->compute();
 
   if (profilingEnabled_)
-    computeTimer_.stop();
+    computeTimer_ += clock();
```
Comment (breznak) [inline src/nupic/engine/Region.cpp]: also, rather getElapsed() from utils/Time.hpp

File: src/test/unit/algorithms/SpatialPoolerTest.cpp (hunk 2 above)
Comment (ctrl-z-9000-times) [reply src/test/unit/algorithms/SpatialPoolerTest.cpp]: Ok, I can rewrite it as follows:
```
double testTimer = 0 // seconds
for(){
  clock_t start_time = clock()
  -do-work-
  testTimer += getElapsed(start_time)
}
```

**Comment by ctrl-z-9000-times** [comment]
I would still like to do this PR but:
* It breaks nupic.py just a little
* Me and @breznak disagree about which clock to use.  My opinions here are not strong, but I've already gone and written the PR using ctime/clock()

So I'm going to put this PR on hold until the python repository situation is sorted out.

**Comment by ctrl-z-9000-times** [comment]
I took another look at this PR, the current Timer code, and what I would need to do for this PR to move forward.  It looks like we're going to use the the C++ `<chrono>` library.  The C++ timers are not as simple to use as the C function `clock()` and so if we're going to use them then it makes sense to use the Timer class.  I'm closing this PR.

**Pull request closed without merging**